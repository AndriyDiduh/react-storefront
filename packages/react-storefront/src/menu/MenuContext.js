/**
 * @license
 * Copyright Â© 2017-2019 Moov Corporation.  All rights reserved.
 */
import React from 'react'
const MenuContext = React.createContext()
export default MenuContext
