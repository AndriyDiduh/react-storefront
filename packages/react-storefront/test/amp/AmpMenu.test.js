/**
 * @license
 * Copyright Â© 2017-2018 Moov Corporation.  All rights reserved.
 */
import React from 'react'
import { mount } from 'enzyme'
import Menu from '../../src/amp/AmpMenu'
import MenuContext from '../../src/menu/MenuContext'
import Provider from '../TestProvider'
import AppModelBase from '../../src/model/AppModelBase'

describe('AmpMenu', () => {
  let app

  beforeEach(() => {
    app = AppModelBase.create({
      amp: true,
      location: {
        pathname: '/',
        search: '',
        hostname: 'localhost'
      },
      menu: {
        levels: [
          {
            root: true,
            items: [
              {
                text: 'Group 1',
                className: 'group-1',
                items: [
                  {
                    text: 'Item 1',
                    url: '/item1',
                    items: [{ text: 'Child 1', url: '/item1/child1' }]
                  }
                ]
              }
            ]
          }
        ]
      }
    })
  })

  it('should render full AMP menu', () => {
    const wrapper = mount(
      <Provider app={app}>
        <MenuContext.Provider value={{ classes: {} }}>
          <Menu />
        </MenuContext.Provider>
      </Provider>
    )

    expect(wrapper).toMatchSnapshot()

    expect(
      wrapper
        .find('ItemContent')
        .at(0)
        .text()
    ).toBe('Group 1')

    expect(
      wrapper
        .find('ItemContent')
        .at(1)
        .text()
    ).toBe('Item 1')

    expect(
      wrapper
        .find('ItemContent')
        .at(2)
        .text()
    ).toBe('Child 1')
  })
})
