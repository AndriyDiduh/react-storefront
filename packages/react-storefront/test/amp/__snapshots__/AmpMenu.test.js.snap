// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AmpMenu should render full AMP menu 1`] = `
<TestProvider app={{...}}>
  <Provider router={{...}} app={{...}} history={{...}} nextId={[Function: nextId]}>
    <MuiThemeProviderOld theme={{...}}>
      <WithTheme(WithStyles(inject-AmpMenu))>
        <WithStyles(inject-AmpMenu) theme={{...}}>
          <inject-AmpMenu theme={{...}} classes={{...}}>
            <AmpMenu theme={{...}} classes={{...}} menu={{...}} id="moov_menu">
              <inject-AmpState id="rsfMenu" initialState={{...}}>
                <AmpState id="rsfMenu" initialState={{...}} amp={true}>
                  <Provider ampStateId="rsfMenu">
                    <HelmetWrapper defer={true} encodeSpecialCharacters={true}>
                      <SideEffect(NullComponent) defer={true} encodeSpecialCharacters={true} script={{...}}>
                        <NullComponent defer={true} encodeSpecialCharacters={true} script={{...}} />
                      </SideEffect(NullComponent)>
                    </HelmetWrapper>
                    <amp-state id="rsfMenu">
                      <script type="application/json" dangerouslySetInnerHTML={{...}} />
                    </amp-state>
                  </Provider>
                </AmpState>
              </inject-AmpState>
              <HelmetWrapper defer={true} encodeSpecialCharacters={true}>
                <SideEffect(NullComponent) defer={true} encodeSpecialCharacters={true} script={{...}} style={{...}}>
                  <NullComponent defer={true} encodeSpecialCharacters={true} script={{...}} style={{...}} />
                </SideEffect(NullComponent)>
              </HelmetWrapper>
              <amp-sidebar id="moov_menu" class="RSFAmpMenu-root-1" layout="nodisplay" side={[undefined]} on="sidebarClose:AMP.setState({ rsfMenu: { open: false } })">
                <MenuBody rootHeader={[undefined]} rootFooter={[undefined]} drawerWidth={[undefined]} ExpandIcon={[undefined]} CollapseIcon={[undefined]} root={{...}} depth={0} path={{...}} classes={{...}} theme={{...}} useExpanders={true}>
                  <div className="" amp-bind="class=>rsfMenu.list == \\'@\\'  ? \\'\\' : \\'\\'">
                    <WithStyles(List) classes={{...}}>
                      <List classes={{...}} component="ul" dense={false} disablePadding={false}>
                        <ul className="MuiList-root-2 MuiList-padding-3">
                          <inject-Item-with-app rootHeader={[undefined]} rootFooter={[undefined]} drawerWidth={[undefined]} ExpandIcon={[undefined]} CollapseIcon={[undefined]} root={{...}} depth={0} path={{...}} classes={{...}} theme={{...}} useExpanders={true} item={{...}} index={0}>
                            <Item rootHeader={[undefined]} rootFooter={[undefined]} drawerWidth={[undefined]} ExpandIcon={[undefined]} CollapseIcon={[undefined]} root={{...}} depth={0} path={{...}} classes={{...}} theme={{...}} useExpanders={true} item={{...}} index={0} app={{...}}>
                              <inject-Branch-with-app expandFirstItem={[undefined]} itemContentRenderer={[undefined]} itemRenderer={[undefined]} trackSelected={[undefined]} ExpandIcon={[undefined]} CollapseIcon={[undefined]} theme={{...}} item={{...}} index={0} depth={0} useExpanders={true} simple={[undefined]}>
                                <Branch expandFirstItem={[undefined]} itemContentRenderer={[undefined]} itemRenderer={[undefined]} trackSelected={[undefined]} ExpandIcon={[undefined]} CollapseIcon={[undefined]} theme={{...}} item={{...}} index={0} depth={0} useExpanders={true} simple={[undefined]} app={{...}}>
                                  <div amp-bind="class=>rsfMenu.sublist == \\'0.0\\' ? \\'expanded\\' : \\'\\'">
                                    <WithStyles(ListItem) className="menu-item" button={true} divider={true} on="tap:AMP.setState({ rsfMenu: { list: \\'@0\\' } })">
                                      <ListItem className="menu-item" button={true} divider={true} on="tap:AMP.setState({ rsfMenu: { list: \\'@0\\' } })" classes={{...}} alignItems="center" ContainerComponent="li" dense={false} disabled={false} disableGutters={false} selected={false}>
                                        <MergeListContext dense={false} alignItems="center">
                                          <WithStyles(ButtonBase) className="MuiListItem-root-6 MuiListItem-default-9 MuiListItem-gutters-14 MuiListItem-divider-13 MuiListItem-button-15 menu-item" disabled={false} on="tap:AMP.setState({ rsfMenu: { list: \\'@0\\' } })" component="div" focusVisibleClassName="MuiListItem-focusVisible-8">
                                            <ButtonBase className="MuiListItem-root-6 MuiListItem-default-9 MuiListItem-gutters-14 MuiListItem-divider-13 MuiListItem-button-15 menu-item" disabled={false} on="tap:AMP.setState({ rsfMenu: { list: \\'@0\\' } })" component="div" focusVisibleClassName="MuiListItem-focusVisible-8" classes={{...}} centerRipple={false} disableRipple={false} disableTouchRipple={false} focusRipple={false} tabIndex="0" type="button">
                                              <div className="MuiButtonBase-root-18 MuiListItem-root-6 MuiListItem-default-9 MuiListItem-gutters-14 MuiListItem-divider-13 MuiListItem-button-15 menu-item" onBlur={[Function]} onFocus={[Function]} onKeyDown={[Function]} onKeyUp={[Function]} onMouseDown={[Function]} onMouseLeave={[Function]} onMouseUp={[Function]} onTouchEnd={[Function]} onTouchMove={[Function]} onTouchStart={[Function]} tabIndex="0" role="button" on="tap:AMP.setState({ rsfMenu: { list: \\'@0\\' } })">
                                                <inject-ItemContent-with-app expandFirstItem={[undefined]} itemContentRenderer={[undefined]} itemRenderer={[undefined]} trackSelected={[undefined]} ExpandIcon={[undefined]} CollapseIcon={[undefined]} theme={{...}} item={{...}} leaf={false} showExpander={false} sublist="0.0">
                                                  <ItemContent expandFirstItem={[undefined]} itemContentRenderer={[undefined]} itemRenderer={[undefined]} trackSelected={[undefined]} ExpandIcon={[undefined]} CollapseIcon={[undefined]} theme={{...}} item={{...}} leaf={false} showExpander={false} sublist="0.0" app={{...}}>
                                                    <WithStyles(ListItemText) className={[undefined]} primary="Group 1" disableTypography={true}>
                                                      <ListItemText className={[undefined]} primary="Group 1" disableTypography={true} classes={{...}} inset={false}>
                                                        <div className="MuiListItemText-root-21">
                                                          Group 1
                                                        </div>
                                                      </ListItemText>
                                                    </WithStyles(ListItemText)>
                                                    <WithStyles(ListItemIcon) className={[undefined]}>
                                                      <ListItemIcon className={[undefined]} classes={{...}}>
                                                        <div className="MuiListItemIcon-root-27">
                                                          <inject-ExpanderIcon-with-app expandFirstItem={[undefined]} itemContentRenderer={[undefined]} itemRenderer={[undefined]} trackSelected={[undefined]} ExpandIcon={[undefined]} CollapseIcon={[undefined]} theme={{...}} item={{...}} leaf={false} showExpander={false} sublist="0.0" app={{...}}>
                                                            <ExpanderIcon expandFirstItem={[undefined]} itemContentRenderer={[undefined]} itemRenderer={[undefined]} trackSelected={[undefined]} ExpandIcon={[undefined]} CollapseIcon={[undefined]} theme={{...}} item={{...}} leaf={false} showExpander={false} sublist="0.0" app={{...}}>
                                                              <pure(ChevronRightIcon) className={[undefined]}>
                                                                <ChevronRightIcon className={[undefined]}>
                                                                  <WithStyles(SvgIcon) className={[undefined]}>
                                                                    <SvgIcon className={[undefined]} classes={{...}} color="inherit" component="svg" fontSize="default" viewBox="0 0 24 24">
                                                                      <svg className="MuiSvgIcon-root-28" focusable="false" viewBox="0 0 24 24" color={[undefined]} aria-hidden="true" role="presentation">
                                                                        <path d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z" />
                                                                        <path fill="none" d="M0 0h24v24H0z" />
                                                                      </svg>
                                                                    </SvgIcon>
                                                                  </WithStyles(SvgIcon)>
                                                                </ChevronRightIcon>
                                                              </pure(ChevronRightIcon)>
                                                            </ExpanderIcon>
                                                          </inject-ExpanderIcon-with-app>
                                                        </div>
                                                      </ListItemIcon>
                                                    </WithStyles(ListItemIcon)>
                                                  </ItemContent>
                                                </inject-ItemContent-with-app>
                                                <NoSsr defer={false} fallback={{...}}>
                                                  <WithStyles(TouchRipple) innerRef={[Function]} center={false}>
                                                    <TouchRipple center={false} classes={{...}}>
                                                      <TransitionGroup component="span" enter={true} exit={true} className="MuiTouchRipple-root-77" childFactory={[Function: childFactory]}>
                                                        <span className="MuiTouchRipple-root-77" />
                                                      </TransitionGroup>
                                                    </TouchRipple>
                                                  </WithStyles(TouchRipple)>
                                                </NoSsr>
                                              </div>
                                            </ButtonBase>
                                          </WithStyles(ButtonBase)>
                                        </MergeListContext>
                                      </ListItem>
                                    </WithStyles(ListItem)>
                                  </div>
                                </Branch>
                              </inject-Branch-with-app>
                            </Item>
                          </inject-Item-with-app>
                        </ul>
                      </List>
                    </WithStyles(List)>
                  </div>
                </MenuBody>
                <MenuBody rootHeader={[undefined]} rootFooter={[undefined]} drawerWidth={[undefined]} ExpandIcon={[undefined]} CollapseIcon={[undefined]} root={{...}} depth={1} path={{...}} classes={{...}} theme={{...}} useExpanders={true}>
                  <div className="undefined" amp-bind="class=>rsfMenu.list == \\'@0\\'  ? \\'\\' : \\'undefined\\'">
                    <WithStyles(List) classes={{...}}>
                      <List classes={{...}} component="ul" dense={false} disablePadding={false}>
                        <ul className="MuiList-root-2 MuiList-padding-3">
                          <WithStyles(ListItem) divider={true} button={true} on="tap:AMP.setState({ rsfMenu: { list: \\'@\\' } })">
                            <ListItem divider={true} button={true} on="tap:AMP.setState({ rsfMenu: { list: \\'@\\' } })" classes={{...}} alignItems="center" ContainerComponent="li" dense={false} disabled={false} disableGutters={false} selected={false}>
                              <MergeListContext dense={false} alignItems="center">
                                <WithStyles(ButtonBase) className="MuiListItem-root-6 MuiListItem-default-9 MuiListItem-gutters-14 MuiListItem-divider-13 MuiListItem-button-15" disabled={false} on="tap:AMP.setState({ rsfMenu: { list: \\'@\\' } })" component="div" focusVisibleClassName="MuiListItem-focusVisible-8">
                                  <ButtonBase className="MuiListItem-root-6 MuiListItem-default-9 MuiListItem-gutters-14 MuiListItem-divider-13 MuiListItem-button-15" disabled={false} on="tap:AMP.setState({ rsfMenu: { list: \\'@\\' } })" component="div" focusVisibleClassName="MuiListItem-focusVisible-8" classes={{...}} centerRipple={false} disableRipple={false} disableTouchRipple={false} focusRipple={false} tabIndex="0" type="button">
                                    <div className="MuiButtonBase-root-18 MuiListItem-root-6 MuiListItem-default-9 MuiListItem-gutters-14 MuiListItem-divider-13 MuiListItem-button-15" onBlur={[Function]} onFocus={[Function]} onKeyDown={[Function]} onKeyUp={[Function]} onMouseDown={[Function]} onMouseLeave={[Function]} onMouseUp={[Function]} onTouchEnd={[Function]} onTouchMove={[Function]} onTouchStart={[Function]} tabIndex="0" role="button" on="tap:AMP.setState({ rsfMenu: { list: \\'@\\' } })">
                                      <WithStyles(ListItemIcon) classes={{...}}>
                                        <ListItemIcon classes={{...}}>
                                          <div className="MuiListItemIcon-root-27">
                                            <pure(ChevronLeftIcon) className={[undefined]}>
                                              <ChevronLeftIcon className={[undefined]}>
                                                <WithStyles(SvgIcon) className={[undefined]}>
                                                  <SvgIcon className={[undefined]} classes={{...}} color="inherit" component="svg" fontSize="default" viewBox="0 0 24 24">
                                                    <svg className="MuiSvgIcon-root-28" focusable="false" viewBox="0 0 24 24" color={[undefined]} aria-hidden="true" role="presentation">
                                                      <path d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z" />
                                                      <path fill="none" d="M0 0h24v24H0z" />
                                                    </svg>
                                                  </SvgIcon>
                                                </WithStyles(SvgIcon)>
                                              </ChevronLeftIcon>
                                            </pure(ChevronLeftIcon)>
                                          </div>
                                        </ListItemIcon>
                                      </WithStyles(ListItemIcon)>
                                      <WithStyles(ListItemText) classes={{...}} primary={{...}}>
                                        <ListItemText primary={{...}} classes={{...}} disableTypography={false} inset={false}>
                                          <div className="MuiListItemText-root-21">
                                            <WithStyles(Typography) variant="subheading" internalDeprecatedVariant={true} className="MuiListItemText-primary-24" component="span">
                                              <Typography variant="subheading" internalDeprecatedVariant={true} className="MuiListItemText-primary-24" component="span" theme={{...}} classes={{...}} align="inherit" color="default" gutterBottom={false} headlineMapping={{...}} noWrap={false} paragraph={false}>
                                                <span className="MuiTypography-root-37 MuiTypography-subtitle1-55 MuiListItemText-primary-24">
                                                  <div className={[undefined]}>
                                                    Group 1
                                                     
                                                  </div>
                                                </span>
                                              </Typography>
                                            </WithStyles(Typography)>
                                          </div>
                                        </ListItemText>
                                      </WithStyles(ListItemText)>
                                      <NoSsr defer={false} fallback={{...}}>
                                        <WithStyles(TouchRipple) innerRef={[Function]} center={false}>
                                          <TouchRipple center={false} classes={{...}}>
                                            <TransitionGroup component="span" enter={true} exit={true} className="MuiTouchRipple-root-77" childFactory={[Function: childFactory]}>
                                              <span className="MuiTouchRipple-root-77" />
                                            </TransitionGroup>
                                          </TouchRipple>
                                        </WithStyles(TouchRipple)>
                                      </NoSsr>
                                    </div>
                                  </ButtonBase>
                                </WithStyles(ButtonBase)>
                              </MergeListContext>
                            </ListItem>
                          </WithStyles(ListItem)>
                          <inject-Item-with-app rootHeader={[undefined]} rootFooter={[undefined]} drawerWidth={[undefined]} ExpandIcon={[undefined]} CollapseIcon={[undefined]} root={{...}} depth={1} path={{...}} classes={{...}} theme={{...}} useExpanders={true} item={{...}} index={0}>
                            <Item rootHeader={[undefined]} rootFooter={[undefined]} drawerWidth={[undefined]} ExpandIcon={[undefined]} CollapseIcon={[undefined]} root={{...}} depth={1} path={{...}} classes={{...}} theme={{...}} useExpanders={true} item={{...}} index={0} app={{...}}>
                              <inject-Branch-with-app expandFirstItem={[undefined]} itemContentRenderer={[undefined]} itemRenderer={[undefined]} trackSelected={[undefined]} ExpandIcon={[undefined]} CollapseIcon={[undefined]} theme={{...}} item={{...}} index={0} depth={1} useExpanders={true} simple={[undefined]}>
                                <Branch expandFirstItem={[undefined]} itemContentRenderer={[undefined]} itemRenderer={[undefined]} trackSelected={[undefined]} ExpandIcon={[undefined]} CollapseIcon={[undefined]} theme={{...}} item={{...}} index={0} depth={1} useExpanders={true} simple={[undefined]} app={{...}}>
                                  <div amp-bind="class=>rsfMenu.sublist == \\'1.0\\' ? \\'expanded\\' : \\'\\'">
                                    <WithStyles(ListItem) className="menu-item" button={true} divider={true} on="tap:AMP.setState({ rsfMenu: { sublist: sublist == \\'1.0\\' ? null : \\'1.0\\' } })">
                                      <ListItem className="menu-item" button={true} divider={true} on="tap:AMP.setState({ rsfMenu: { sublist: sublist == \\'1.0\\' ? null : \\'1.0\\' } })" classes={{...}} alignItems="center" ContainerComponent="li" dense={false} disabled={false} disableGutters={false} selected={false}>
                                        <MergeListContext dense={false} alignItems="center">
                                          <WithStyles(ButtonBase) className="MuiListItem-root-6 MuiListItem-default-9 MuiListItem-gutters-14 MuiListItem-divider-13 MuiListItem-button-15 menu-item" disabled={false} on="tap:AMP.setState({ rsfMenu: { sublist: sublist == \\'1.0\\' ? null : \\'1.0\\' } })" component="div" focusVisibleClassName="MuiListItem-focusVisible-8">
                                            <ButtonBase className="MuiListItem-root-6 MuiListItem-default-9 MuiListItem-gutters-14 MuiListItem-divider-13 MuiListItem-button-15 menu-item" disabled={false} on="tap:AMP.setState({ rsfMenu: { sublist: sublist == \\'1.0\\' ? null : \\'1.0\\' } })" component="div" focusVisibleClassName="MuiListItem-focusVisible-8" classes={{...}} centerRipple={false} disableRipple={false} disableTouchRipple={false} focusRipple={false} tabIndex="0" type="button">
                                              <div className="MuiButtonBase-root-18 MuiListItem-root-6 MuiListItem-default-9 MuiListItem-gutters-14 MuiListItem-divider-13 MuiListItem-button-15 menu-item" onBlur={[Function]} onFocus={[Function]} onKeyDown={[Function]} onKeyUp={[Function]} onMouseDown={[Function]} onMouseLeave={[Function]} onMouseUp={[Function]} onTouchEnd={[Function]} onTouchMove={[Function]} onTouchStart={[Function]} tabIndex="0" role="button" on="tap:AMP.setState({ rsfMenu: { sublist: sublist == \\'1.0\\' ? null : \\'1.0\\' } })">
                                                <inject-ItemContent-with-app expandFirstItem={[undefined]} itemContentRenderer={[undefined]} itemRenderer={[undefined]} trackSelected={[undefined]} ExpandIcon={[undefined]} CollapseIcon={[undefined]} theme={{...}} item={{...}} leaf={false} showExpander={true} sublist="1.0">
                                                  <ItemContent expandFirstItem={[undefined]} itemContentRenderer={[undefined]} itemRenderer={[undefined]} trackSelected={[undefined]} ExpandIcon={[undefined]} CollapseIcon={[undefined]} theme={{...}} item={{...}} leaf={false} showExpander={true} sublist="1.0" app={{...}}>
                                                    <WithStyles(ListItemText) className={[undefined]} primary="Item 1" disableTypography={true}>
                                                      <ListItemText className={[undefined]} primary="Item 1" disableTypography={true} classes={{...}} inset={false}>
                                                        <div className="MuiListItemText-root-21">
                                                          Item 1
                                                        </div>
                                                      </ListItemText>
                                                    </WithStyles(ListItemText)>
                                                    <WithStyles(ListItemIcon) className={[undefined]}>
                                                      <ListItemIcon className={[undefined]} classes={{...}}>
                                                        <div className="MuiListItemIcon-root-27">
                                                          <inject-ExpanderIcon-with-app expandFirstItem={[undefined]} itemContentRenderer={[undefined]} itemRenderer={[undefined]} trackSelected={[undefined]} ExpandIcon={[undefined]} CollapseIcon={[undefined]} theme={{...}} item={{...}} leaf={false} showExpander={true} sublist="1.0" app={{...}}>
                                                            <ExpanderIcon expandFirstItem={[undefined]} itemContentRenderer={[undefined]} itemRenderer={[undefined]} trackSelected={[undefined]} ExpandIcon={[undefined]} CollapseIcon={[undefined]} theme={{...}} item={{...}} leaf={false} showExpander={true} sublist="1.0" app={{...}}>
                                                              <pure(ExpandLessIcon) className={[undefined]} amp-bind="class=>rsfMenu.sublist == \\'1.0\\' ? \\'undefined undefined\\' : \\'undefined undefined\\'">
                                                                <ExpandLessIcon className={[undefined]} amp-bind="class=>rsfMenu.sublist == \\'1.0\\' ? \\'undefined undefined\\' : \\'undefined undefined\\'">
                                                                  <WithStyles(SvgIcon) className={[undefined]} amp-bind="class=>rsfMenu.sublist == \\'1.0\\' ? \\'undefined undefined\\' : \\'undefined undefined\\'">
                                                                    <SvgIcon className={[undefined]} amp-bind="class=>rsfMenu.sublist == \\'1.0\\' ? \\'undefined undefined\\' : \\'undefined undefined\\'" classes={{...}} color="inherit" component="svg" fontSize="default" viewBox="0 0 24 24">
                                                                      <svg className="MuiSvgIcon-root-28" focusable="false" viewBox="0 0 24 24" color={[undefined]} aria-hidden="true" role="presentation" amp-bind="class=>rsfMenu.sublist == \\'1.0\\' ? \\'undefined undefined\\' : \\'undefined undefined\\'">
                                                                        <path d="M12 8l-6 6 1.41 1.41L12 10.83l4.59 4.58L18 14z" />
                                                                        <path fill="none" d="M0 0h24v24H0z" />
                                                                      </svg>
                                                                    </SvgIcon>
                                                                  </WithStyles(SvgIcon)>
                                                                </ExpandLessIcon>
                                                              </pure(ExpandLessIcon)>
                                                              <pure(ExpandMoreIcon) className={[undefined]} amp-bind="class=>rsfMenu.sublist == \\'1.0\\' ? \\'undefined undefined\\' : \\'undefined undefined\\'">
                                                                <ExpandMoreIcon className={[undefined]} amp-bind="class=>rsfMenu.sublist == \\'1.0\\' ? \\'undefined undefined\\' : \\'undefined undefined\\'">
                                                                  <WithStyles(SvgIcon) className={[undefined]} amp-bind="class=>rsfMenu.sublist == \\'1.0\\' ? \\'undefined undefined\\' : \\'undefined undefined\\'">
                                                                    <SvgIcon className={[undefined]} amp-bind="class=>rsfMenu.sublist == \\'1.0\\' ? \\'undefined undefined\\' : \\'undefined undefined\\'" classes={{...}} color="inherit" component="svg" fontSize="default" viewBox="0 0 24 24">
                                                                      <svg className="MuiSvgIcon-root-28" focusable="false" viewBox="0 0 24 24" color={[undefined]} aria-hidden="true" role="presentation" amp-bind="class=>rsfMenu.sublist == \\'1.0\\' ? \\'undefined undefined\\' : \\'undefined undefined\\'">
                                                                        <path d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z" />
                                                                        <path fill="none" d="M0 0h24v24H0z" />
                                                                      </svg>
                                                                    </SvgIcon>
                                                                  </WithStyles(SvgIcon)>
                                                                </ExpandMoreIcon>
                                                              </pure(ExpandMoreIcon)>
                                                            </ExpanderIcon>
                                                          </inject-ExpanderIcon-with-app>
                                                        </div>
                                                      </ListItemIcon>
                                                    </WithStyles(ListItemIcon)>
                                                  </ItemContent>
                                                </inject-ItemContent-with-app>
                                                <NoSsr defer={false} fallback={{...}}>
                                                  <WithStyles(TouchRipple) innerRef={[Function]} center={false}>
                                                    <TouchRipple center={false} classes={{...}}>
                                                      <TransitionGroup component="span" enter={true} exit={true} className="MuiTouchRipple-root-77" childFactory={[Function: childFactory]}>
                                                        <span className="MuiTouchRipple-root-77" />
                                                      </TransitionGroup>
                                                    </TouchRipple>
                                                  </WithStyles(TouchRipple)>
                                                </NoSsr>
                                              </div>
                                            </ButtonBase>
                                          </WithStyles(ButtonBase)>
                                        </MergeListContext>
                                      </ListItem>
                                    </WithStyles(ListItem)>
                                  </div>
                                  <WithStyles(Collapse) in={true} amp-bind="class=>rsfMenu.sublist == \\'1.0\\' ? \\'undefined\\' : \\'undefined\\'" timeout="auto">
                                    <Collapse in={true} amp-bind="class=>rsfMenu.sublist == \\'1.0\\' ? \\'undefined\\' : \\'undefined\\'" timeout="auto" theme={{...}} classes={{...}} collapsedHeight="0px" component="div">
                                      <Transition onEnter={[Function]} onEntered={[Function]} onEntering={[Function]} onExit={[Function]} onExiting={[Function]} addEndListener={[Function]} timeout={{...}} in={true} amp-bind="class=>rsfMenu.sublist == \\'1.0\\' ? \\'undefined\\' : \\'undefined\\'" mountOnEnter={false} unmountOnExit={false} appear={false} enter={true} exit={true} onExited={[Function: noop]}>
                                        <div className="MuiCollapse-container-72 MuiCollapse-entered-73" style={{...}} amp-bind="class=>rsfMenu.sublist == \\'1.0\\' ? \\'undefined\\' : \\'undefined\\'">
                                          <div className="MuiCollapse-wrapper-74">
                                            <div className="MuiCollapse-wrapperInner-75">
                                              <WithStyles(List) component="div" classes={{...}}>
                                                <List component="div" classes={{...}} dense={false} disablePadding={false}>
                                                  <div className="MuiList-root-2 MuiList-padding-3">
                                                    <inject-Item-with-app expandFirstItem={[undefined]} itemContentRenderer={[undefined]} itemRenderer={[undefined]} trackSelected={[undefined]} ExpandIcon={[undefined]} CollapseIcon={[undefined]} theme={{...}} item={{...}} index={0} depth={2} useExpanders={true} simple={[undefined]} app={{...}}>
                                                      <Item expandFirstItem={[undefined]} itemContentRenderer={[undefined]} itemRenderer={[undefined]} trackSelected={[undefined]} ExpandIcon={[undefined]} CollapseIcon={[undefined]} theme={{...}} item={{...}} index={0} depth={2} useExpanders={true} simple={[undefined]} app={{...}}>
                                                        <inject-Leaf-with-app expandFirstItem={[undefined]} itemContentRenderer={[undefined]} itemRenderer={[undefined]} trackSelected={[undefined]} ExpandIcon={[undefined]} CollapseIcon={[undefined]} theme={{...}} item={{...}} index={0} depth={2} useExpanders={true} simple={[undefined]}>
                                                          <Leaf expandFirstItem={[undefined]} itemContentRenderer={[undefined]} itemRenderer={[undefined]} trackSelected={[undefined]} ExpandIcon={[undefined]} CollapseIcon={[undefined]} theme={{...}} item={{...}} index={0} depth={2} useExpanders={true} simple={[undefined]} app={{...}}>
                                                            <WithStyles(inject-Link) to="/item1/child1" className={[undefined]} server={false} state={{...}} onClick={[Function]}>
                                                              <inject-Link to="/item1/child1" className={[undefined]} server={false} state={{...}} onClick={[Function]} classes={{...}}>
                                                                <Link to="/item1/child1" className={[undefined]} server={false} state={{...}} onClick={[Function]} classes={{...}} history={{...}} router={{...}} app={{...}} anchorProps={{...}} preserveScroll={false}>
                                                                  <a data-moov-link="on" className="RSFLink-root-76" style={[undefined]} onClick={[Function]} href="https://localhost/item1/child1">
                                                                    <WithStyles(ListItem) button={true} divider={true} selected={[undefined]} classes={{...}}>
                                                                      <ListItem button={true} divider={true} selected={false} classes={{...}} alignItems="center" ContainerComponent="li" dense={false} disabled={false} disableGutters={false}>
                                                                        <MergeListContext dense={false} alignItems="center">
                                                                          <WithStyles(ButtonBase) className="MuiListItem-root-6 MuiListItem-default-9 MuiListItem-gutters-14 MuiListItem-divider-13 MuiListItem-button-15" disabled={false} component="div" focusVisibleClassName="MuiListItem-focusVisible-8">
                                                                            <ButtonBase className="MuiListItem-root-6 MuiListItem-default-9 MuiListItem-gutters-14 MuiListItem-divider-13 MuiListItem-button-15" disabled={false} component="div" focusVisibleClassName="MuiListItem-focusVisible-8" classes={{...}} centerRipple={false} disableRipple={false} disableTouchRipple={false} focusRipple={false} tabIndex="0" type="button">
                                                                              <div className="MuiButtonBase-root-18 MuiListItem-root-6 MuiListItem-default-9 MuiListItem-gutters-14 MuiListItem-divider-13 MuiListItem-button-15" onBlur={[Function]} onFocus={[Function]} onKeyDown={[Function]} onKeyUp={[Function]} onMouseDown={[Function]} onMouseLeave={[Function]} onMouseUp={[Function]} onTouchEnd={[Function]} onTouchMove={[Function]} onTouchStart={[Function]} tabIndex="0" role="button">
                                                                                <inject-ItemContent-with-app expandFirstItem={[undefined]} itemContentRenderer={[undefined]} itemRenderer={[undefined]} ExpandIcon={[undefined]} CollapseIcon={[undefined]} theme={{...}} index={0} depth={2} useExpanders={true} simple={[undefined]} item={{...}} showExpander={false} leaf={true}>
                                                                                  <ItemContent expandFirstItem={[undefined]} itemContentRenderer={[undefined]} itemRenderer={[undefined]} ExpandIcon={[undefined]} CollapseIcon={[undefined]} theme={{...}} index={0} depth={2} useExpanders={true} simple={[undefined]} item={{...}} showExpander={false} leaf={true} app={{...}}>
                                                                                    <WithStyles(ListItemText) primary="Child 1" disableTypography={true}>
                                                                                      <ListItemText primary="Child 1" disableTypography={true} classes={{...}} inset={false}>
                                                                                        <div className="MuiListItemText-root-21">
                                                                                          Child 1
                                                                                        </div>
                                                                                      </ListItemText>
                                                                                    </WithStyles(ListItemText)>
                                                                                  </ItemContent>
                                                                                </inject-ItemContent-with-app>
                                                                                <NoSsr defer={false} fallback={{...}}>
                                                                                  <WithStyles(TouchRipple) innerRef={[Function]} center={false}>
                                                                                    <TouchRipple center={false} classes={{...}}>
                                                                                      <TransitionGroup component="span" enter={true} exit={true} className="MuiTouchRipple-root-77" childFactory={[Function: childFactory]}>
                                                                                        <span className="MuiTouchRipple-root-77" />
                                                                                      </TransitionGroup>
                                                                                    </TouchRipple>
                                                                                  </WithStyles(TouchRipple)>
                                                                                </NoSsr>
                                                                              </div>
                                                                            </ButtonBase>
                                                                          </WithStyles(ButtonBase)>
                                                                        </MergeListContext>
                                                                      </ListItem>
                                                                    </WithStyles(ListItem)>
                                                                  </a>
                                                                </Link>
                                                              </inject-Link>
                                                            </WithStyles(inject-Link)>
                                                          </Leaf>
                                                        </inject-Leaf-with-app>
                                                      </Item>
                                                    </inject-Item-with-app>
                                                  </div>
                                                </List>
                                              </WithStyles(List)>
                                            </div>
                                          </div>
                                        </div>
                                      </Transition>
                                    </Collapse>
                                  </WithStyles(Collapse)>
                                </Branch>
                              </inject-Branch-with-app>
                            </Item>
                          </inject-Item-with-app>
                        </ul>
                      </List>
                    </WithStyles(List)>
                  </div>
                </MenuBody>
              </amp-sidebar>
            </AmpMenu>
          </inject-AmpMenu>
        </WithStyles(inject-AmpMenu)>
      </WithTheme(WithStyles(inject-AmpMenu))>
    </MuiThemeProviderOld>
  </Provider>
</TestProvider>
`;
